src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -107,11 +107,23 @@ class graph_impl {
     device->update_graph_rtp(handle, port, buffer, size);
   }
 
+  void
+  update_rtp(const std::string& port, const char* buffer, size_t size)
+  {
+    device->update_graph_rtp(handle, port, buffer, size);
+  }
+
   void
   read_rtp(const char* port, char* buffer, size_t size)
   {
     device->read_graph_rtp(handle, port, buffer, size);
   }
+
+  void
+  read_rtp(const std::string& port, char* buffer, size_t size)
+  {
+    device->read_graph_rtp(handle, port, buffer, size);
+  }
```

Comment:
Is there an reason to add the shim read_graph_rtp for the std::string version?   For convenience it may make sense to keep read_rtp for std::string, but maybe it should simple call read_rtp(port.c_str(), buffer, size)?   

If on the other hand the const char* version of read_rtp ends up converting the port to a std::string at a lower level then isn't read_rtp(const std::stirng&, ...) sufficient?

Reply:
It is on my todo list to use char* to replace std::string& for port name in the rtp_type unordered map and function in xcl level rtp update/read to increase rtp performance by skipping the string allocator. That is why I keep read_trp(const char * ...). I have thought since the c++ api already take std::string (C api takes char *), we should keep using std::string.

But rethink about it, I agree with you, read_rtp(port.c_str(),...) should just work. I will remove the read_rtp(std::string&...) version

Reply:
std::string& functions are removed.